website/addons/base/__init__.py

```diff
@@ -595,7 +595,7 @@ def get_nodes_with_oauth_grants(self, external_account):
         return (
             Node.load(node_id)
             for node_id, grants in self.oauth_grants.iteritems()
-            if external_account._id in grants.keys()
+            if (external_account._id in grants.keys() and not Node.load(node_id).is_deleted)
```

Comment:
Made this change. @sloria is it safe to load the node_id just to check if it is deleted?